### DISCUSSION TITLE
Cant get smtp to work

### DISCUSSION THREAD

**Discussion by ezegelmi** [healthchecks/healthchecks#662] [General]
Hello, im using the official docker image on a test server, and i cant get ping email checks to work. I tried running the docker compose up command as is, tried with that and also running additionally docker-compose run web /opt/healthchecks/manage.py smtpd --port 25. I tried opening ports in firewall (wich is disabled by default anyways). Is there something im missing? Im using my own domain with A and MX records of course, DNS is not a problem as it resolves ok. But when i try to telnet port 25 or 2525 on the server i get a refused connection. Do i have to set up a reverse proxy for this?

**Comment by cuu508** [comment]
Hi @ezegelmi, I just [updated](https://github.com/healthchecks/healthchecks/commit/7912f1e4df52a91b88d72081a6ed351caa41ea86) the dockerfile to run `./manage.py smtpd` automatically, same as `./manage.py sendalerts` and `./manage.py sendreports`.

For it to be accessible from the outside, you must also uncomment a line the docker-compose.yml file:

    ports:
        - "8000:8000"
        # Uncomment to enable SMTP on port 25 for incoming email
        # - "25:25"

On the docker host, you can test the SMTP service with telnet. The happy case:

```
telnet 127.0.0.1 25
Trying 127.0.0.1...
Connected to 127.0.0.1.
Escape character is '^]'.
220 0321862a6700 Python SMTP proxy version 0.3
```

And here something is wrong:

```
telnet 127.0.0.1 25
Trying 127.0.0.1...
telnet: Unable to connect to remote host: Connection refused
```

If you can connect to the SMTP service from the docker host, but not from remote machines, then that's likely a firewall problem.
